custom_components/hildebrand_glow_ihd_mqtt/sensor.py

```diff
@@ -253,8 +253,8 @@
     "unit_of_measurement": "GBP",
     "state_class": SensorStateClass.TOTAL_INCREASING,
     "icon": "mdi:cash",
-    "func": lambda js : round(js['gasmeter']['energy']['import']['price']['standingcharge'] + \
-       (js['gasmeter']['energy']['import']['day'] * js['gasmeter']['energy']['import']['price']['unitrate']), 2),
+    "func": lambda js : round((js['gasmeter']['energy']['import']['price']['standingcharge'] or 0)+ \
+       ((js['gasmeter']['energy']['import']['day'] or 0) * (js['gasmeter']['energy']['import']['price']['unitrate'] or 0)), 2),
```

Comment:
Would creating a standalone function to do the calculation and type checking be a bit neater perhaps?

Something along the lines of:

```py
"func": lambda js : calculate_cost(
      js['gasmeter']['energy']['import']['price']['standingcharge'],
      js['gasmeter']['energy']['import']['day'],
      js['gasmeter']['energy']['import']['price']['unitrate']
    )
```
and 

```py
def calculate_cost(standing_charge, day_import, unit_rate):
  return round((standing_charge or 0) + ((day_import or 0) * (unit_rate or 0)), 2)
```

This can also be re-used for the electricity costs too which will probably suffer the same issue when the situation is reversed (have a gas meter, but no electricity meter).